packages/core/src/components/essentials/tooltip.ts

```diff
@@ -214,7 +218,30 @@ export class Tooltip extends Component {
 			"tooltip",
 			"enabled"
 		);
+		// remove tooltip eventListener
+		this.removeTooltipEventListener();
 		if (isTooltipEnabled) {
+			// Grab the tooltip element
+			const holder = select(this.services.domUtils.getHolder());
+			const chartprefix = Tools.getProperty(
+				this.model.getOptions(),
+				"style",
+				"prefix"
+			);
+			this.tooltip = DOMUtils.appendOrSelect(
+				holder,
+				`div.${settings.prefix}--${chartprefix}--tooltip`
+			);
+
+			// Apply html content to the tooltip
+			this.tooltipTextContainer = DOMUtils.appendOrSelect(
+				this.tooltip,
+				"div.content-box"
+			);
+			this.tooltip.style("max-width", null);
+			this.removeTooltipEventListener();
+			this.addTooltipEventListener();
```

Comment:
so why do we need to remove & add every time? Seems like we just need to store a boolean within the class to keep track of if event listeners are actively attached, and if they are then we don't need to keep removing & adding right?

Reply:
`isEventListenerAdded` is now added as a flag to keep track of whether event listeners are actively attached or not.